server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -455,20 +466,39 @@ export function configureWebSocketServices(
 						socket.emit("nack", "", [nackMessage]);
 						return;
 					}
-					contentBatches.forEach((contentBatch) => {
-						const contents = Array.isArray(contentBatch)
-							? contentBatch
-							: [contentBatch];
 
-						for (const content of contents) {
-							const signalMessage: ISignalMessage = {
-								clientId,
-								content,
-							};
+					if (supportedFeaturesMap.get(clientId)?.submit_signals_v2) {
+						for (const signal of contentBatches) {
+							if (isSentSignalMessage(signal)) {
```

Comment:
signal in this case cannot be Array, as is the case with nonTargetted signals below?

Reply:
The real distinction is between v1 signals and v2 signals. 

For original/v1 signals, contentBatches was previously typed as `IDocumentMessage[] | IDocumentMessage[][]`. We now know that IDocumentMessage type was wrong, they are actually JSON strings, but for old signals we still check if elements of contentBatches are individual elements or arrays. 

All v2 signals are of type `ISentSignalMessage[]`. So elements of contentBatches in this case should only be ISentSignalMessage.